### PR TITLE
Remove DoNotContact entities from memory when already used.

### PR DESCRIPTION
**Pull request by biozshock**
| Q                                      | A
| -------------------------------------- | ---
| New feature/enhancement? (use the a.x branch)      | [ x ]

#### Description:
Replace `clear` with `detachEntities` for https://github.com/mautic/mautic/pull/12226

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Send an email to Lead to the non-existing address.
3. Observe DoNotContact is added.

### CODE REVIEW

**Review by mabumusa1 — DISMISSED** [review]
Changes work as expected.

File: app/bundles/LeadBundle/Model/DoNotContact.php [hunk 1]
```diff
@@ -144,7 +154,7 @@ public function isContactable(Lead $contact, $channel)
             $channel = key($channel);
         }
 
-        /** @var \Mautic\LeadBundle\Entity\DoNotContact[] $entries */
+        /** @var DNC[] $entries */
```
Comment (escopecz) [inline app/bundles/LeadBundle/Model/DoNotContact.php]:
```suggestion
        /** @var Collection<int, DNC> $dncEntries */
```

**Review by escopecz — CHANGES_REQUESTED** [review]
I see no other issue except the wrong type hint

File: app/bundles/LeadBundle/Model/DoNotContact.php (hunk 1 above)
Comment (biozshock) [reply app/bundles/LeadBundle/Model/DoNotContact.php]: @escopecz  Removed the annotation, as it's available in the method, which is called.

**Review by escopecz — APPROVED** [review]
Looks good, thanks 👍

**Pull request merged**